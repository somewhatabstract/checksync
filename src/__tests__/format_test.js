// @flow
import Format from "../format";

describe("Format", () => {
    describe("#info", () => {
        it("should prefix with _INFO__", () => {
            // Arrange

            // Act
            const result = Format.info("Test");

            // Assert
            expect(result).toBe(" INFO  Test");
        });
    });

    describe("#error", () => {
        it("should prefix with _ERROR__", () => {
            // Arrange

            // Act
            const result = Format.error("Test");

            // Assert
            expect(result).toBe(" ERROR  Test");
        });
    });

    describe("#warn", () => {
        it("should prefix with _WARNING__", () => {
            // Arrange

            // Act
            const result = Format.warn("Test");

            // Assert
            expect(result).toBe(" WARNING  Test");
        });
    });

    describe("#violation", () => {
        it("should prefix with _MISMATCH__", () => {
            // Arrange

            // Act
            const result = Format.violation("Test");

            // Assert
            expect(result).toBe(" MISMATCH  Test");
        });
    });
});
