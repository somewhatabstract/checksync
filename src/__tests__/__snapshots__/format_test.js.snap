// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Format #code should render with chalk colors 1`] = `"[1m[33mTest[39m[22m"`;

exports[`Format #cwdFilePath should render with chalk colors 1`] = `"[90mTest[39m"`;

exports[`Format #error should prefix with _ERROR__ 1`] = `"    Error  Test"`;

exports[`Format #error should render with chalk colors 1`] = `"[31m    Error [39m Test"`;

exports[`Format #fix should prefix with _FIX_ 1`] = `"      Fix  Test"`;

exports[`Format #fix should render with chalk colors 1`] = `"[1m[92m      Fix [39m[22m Test"`;

exports[`Format #heading should render with chalk colors 1`] = `"[1m[32mTest[39m[22m"`;

exports[`Format #info should prefix with _INFO__ 1`] = `"     Info  Test"`;

exports[`Format #info should render with chalk colors 1`] = `"[34m     Info [39m Test"`;

exports[`Format #mismatch should prefix with _MISMATCH__ 1`] = `" Mismatch  Test"`;

exports[`Format #mismatch should render with chalk colors 1`] = `"[1m[93m Mismatch [39m[22m Test"`;

exports[`Format #verbose should prefix with Verbose 1`] = `"  Verbose  Test"`;

exports[`Format #verbose should render with chalk colors 1`] = `"[90m  Verbose [39m [2mTest[22m"`;

exports[`Format #warn should prefix with _WARNING__ 1`] = `"  Warning  Test"`;

exports[`Format #warn should render with chalk colors 1`] = `"[33m  Warning [39m Test"`;
